tools/bumbershoot/idpqonvert/idpqonvert.xml

```diff
@@ -0,0 +1,228 @@
+        #set $input_name = $input.display_name
+        #set $output_name = $input_name.split(".")[0] + ".idpDB"
+        ln -s '$input' '${input_name}';
+        ln -s '$input_database' '${db_name}';
+
+        idpQonvert
+            -DecoyPrefix '${decoy_prefix}'
+            -WriteQonversionDetails true
+            -ProteinDatabase '${db_name}'
+            -MaxFDR $max_fdr
+            -EmbedSpectrumScanTimes false
+            -EmbedSpectrumSources false
+            #if $scoring.override_scoring
+                #set $score_info = []
+                #for $engine in ["myrimatch:mvh", "xcorr", "sequest:xcorr", "sequest:deltacn", "mascot:score", "x!tandem:expect", "x!tandem:hyperscore", "ms-gf:specevalue"]
+                    #set $name = $engine.replace(":", "_").replace("!","")
+                    #set $weight = $getVar("scoring." + $name + "." + $name + "_weight.value")
+                    #set $type = $getVar("scoring." + $name + "." + $name + "_type.value")
+                    #set $score_info = $score_info + [str($weight) + " " + $type + " " + $engine]
+                    #continue
+                #end for
+                #for $i, $s in enumerate($scoring.extra_score)
+                    #set $name = $s.extra_score_name
+                    #set $type = $s.extra_score_normalization
+                    #set $weight = $s.extra_score_weight
+                    #set $score_info = $score_info + [str($weight) + " " + $type + " " + $name]
+                #end for
+                #set $score_str = "; ".join($score_info)
+                -ScoreInfo '$score_str'
+            #end if
+            #if $advanced_options.use_advanced_options
+                -MaxImportFDR $advanced_options.max_import_fdr
+                -MaxResultRank $advanced_options.max_result_rank
+                -RerankMatches $advanced_options.rerank_matches
+                -Gamma $advanced_options.gamma
+                -Nu $advanced_options.nu
+                -PolynomialDegree $advanced_options.polynomial_degree
+                -TruePositiveThreshold $advanced_options.true_positive_threshold
+                -MaxTrainingRank $advanced_options.max_training_rank
+                -SVMType $advanced_options.svm_type
+                -Kernel $advanced_options.kernel
+                -ChargeStateHandling $advanced_options.charge_state_handling
+                -QonverterMethod $advanced_options.qonverter_method
+                -MinPartitionSize $advanced_options.min_partition_size
+            #end if
+            '${input_name}';
+        mv '$output_name' output
+]]>
+    </command>
+    <inputs>
+        <param name="input" type="data" format="mzid,raw_pepxml" label="Input mzIdentML/pepXML file(s)" />
+        <param name="input_database" format="fasta" type="data" label="Protein Database"/>
+        <param name="decoy_prefix" type="text" label="Decoy Prefix"/>
+        <param name="max_fdr" type="float" label="Max FDR" value="0.05" />
```

Comment:
Hmm, not really. If I add help attributes, they won't be entirely the same between different tools. I would prefer to use macros only when there is a significant savings in space or complexity of updating it.